internal/services/network/public_ip_resource.go

```diff
@@ -73,6 +73,9 @@ func resourcePublicIp() *pluginsdk.Resource {
 					string(network.DdosSettingsProtectionModeVirtualNetworkInherited),
 				}, false),
 				Default: string(network.DdosSettingsProtectionModeVirtualNetworkInherited),
+				DiffSuppressFunc: func(_, old, new string, _ *pluginsdk.ResourceData) bool {
+					return old == "" && new == string(network.DdosSettingsProtectionModeVirtualNetworkInherited)
+				},
```

Comment:
Rather than adding a workaround, we should instead look to set the appropriate default value in the Read function when this is not returned but implied - as such can we look to update this to default this value in the Read function if it's not returned from the API?

We should also raise an API bug over this, since the API should be returning the value here if it's implied from a previous version?

Reply:
This makes sense. I have changed code.